core/services/relay/evm/evm.go

```diff
@@ -134,30 +135,54 @@ func newConfigProvider(lggr logger.Logger, chainSet evm.ChainSet, args relaytype
 	}, nil
 }
 
-func newContractTransmitter(lggr logger.Logger, rargs relaytypes.RelayArgs, transmitterID string, configWatcher *configWatcher) (*ContractTransmitter, error) {
+func newContractTransmitter(lggr logger.Logger, rargs relaytypes.RelayArgs, transmitterID string, configWatcher *configWatcher, pr pipeline.Runner) (*ContractTransmitter, error) {
```

Comment:
not sure if this behaviour is obvious, that depending on pr being nil or not different transmitters are created. wdyt about a separate function all together `newPipelineContractTransmitter` ?

Reply:
Yeah, looks not really good, fixed